xonsh/ptk_shell/completer.py

```diff
@@ -99,6 +101,35 @@ def get_completions(self, document, complete_event):
                 disp = comp[pre:].strip("'\"")
                 yield Completion(comp, -plen, display=disp)
 
+    def __find_begidx(self, line, endidx):
+        if len(line[0:endidx]) == 0:
```

Comment:
I think an equivalent check here is that `endidx <=1`

Reply:
i think you mean `endidx < 1`

Reply:
python indexing (with positive integers) is `[inclusive: exclusive)`

Reply:
yeah, if fell for that as well - indexing starts at 0 so `[0:1]` gives one character